### ISSUE TITLE
Standardize function output in m2t

### ISSUE THREAD

**Issue by miscco** [matlab2tikz/matlab2tikz#831]
So i went scanning the code and i found a lot of different ways functions wokr.

Some return a str, some write directly into the file via the fid, some  return [m2t, str]

I am sure, that this is due to historical growth. If there is some agreement, i would tackle that, to unify the way we return from functions and order the functions depending on their role.

**Comment by egeerardyn** [comment]
@miscco I agree that there is currently little convention and the scope jumps a bit up and down.
This is all related to #477, #445 and #566 as well.

Your idea is very good, but let's wait a moment.
This afternoon, I have been looking into #393, for which I have already uniformized quite a lot of these handlers to
```matlab
[m2t, str] = handler(m2t, h, custom)
```
where `custom` is a newly introduced thingy.
There are a few places where this is not yet possible, most importantly:
 - `handleAllChildren` since that needs to return a `cellstr`
 - `drawHggroup` (just because it adds no value by itself)
 - `drawAxes` (i.e. second highest level)
 - `saveToFile` (i.e. highest level)

But I will make my experimental code public in a short while.

In particular, the last two need a lot of TLC to uniformize them: they build a faux object in `m2t` and certainly `saveToFile` writes the structure out to file. I think it would make sense in the mid-long run (say, 2.0 release) to make sure that those functions live by that interface and the concerns get separated in `drawAxes` and `saveToFile`, i.e.:
 - `drawAxes` should build a str as well
 - `saveToFile` should be split into:
     * `drawFigure` that builds the string representation, and
     * `saveToFile` that does the writing to disk.

The problem with migrating right away is that we use `m2t` as a kind of god object (global variable, or whatever you like to call that) as discussed in the other issue reports. This makes for subtle coupling that is not trivial to untangle. IMHO, the best moment to tackle such invasive operations, is to wait until there are no PRs open since it tends to produce a lot of merge conflicts.
